src/api.ts

```diff
@@ -915,13 +915,55 @@ export function fernsRouter<T>(
     // updates. We won't wind up with corrupted data, just an API error.
     try {
       Object.assign(doc, body);
+      doc.markModified(field);
       await doc.save();
     } catch (error: any) {
-      throw new APIError({
-        title: `PATCH Pre Update error on ${req.params.id}: ${error.message}`,
-        status: 400,
-        error,
-      });
+      if (error instanceof mongoose.Error.VersionError) {
```

Comment:
My main concern with this is it's just disabling the version errors with extra steps. Retry is very reasonable, or even ignoring versioning all together for these types of patches, but only if we look at what the two writes are. If they're both to the same field, then it's still a version error and we should leave it to the client to figure it out. If it's to two different fields, totally fine. The hard part is I'm not sure how you can look at that with how ferns is built.